### PR TITLE
297 saveoutputtotextfile ddt wide rollout

### PR DESCRIPTION
**Pull request by CAfitech**
Bugs/Issues:

Multiple files did not incoprate the saveoutputtotextfileV2 feature
Gyoithon.tsx did not have saveouputotextfileV2 feature.
netcat.tsx did not have saveouputotextfileV2 feature.
RTsort.tsx did not have saveouputotextfileV2 feature.
SMB Enumeration.tsx did not have saveouputotextfileV2 feature.
SearchSploit.tsx did not have saveouputotextfileV2 feature.
Snmap.tsx did not have saveouputotextfileV2 feature.
Traceroute.tsx did not have saveouputotextfileV2 feature.
WPScan.tsx did not have saveouputotextfileV2 feature.
Zerologon.tsx did not have saveouputotextfileV2 feature.
Metagoofile.tsx did have saveouputotextfileV2 feature.
msfvenom did not have saveouputotextfileV2 feature.
theharvestor did not have saveouputotextfileV2 feature.

Fixes:
All these files were fixed and modified with additional code to incorporate the new saveouttotextfileV2 feature. (These are my Aafiya Irfan's Contributions)

I have made a new pull request after adding the yarn.lock file.

### CODE REVIEW

**Comment by bradleybogg** [comment]
Hi @CAfitech @hhenryDeakin 

Thanks for adding these changes. In a professional setting it is highly likely that this pull request would be simply declined.
I'll share some resources with you that are pivotal in understanding how to use Git (incl GitHub) and pull requests that will benefit greatly in your career.

https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/
https://www.freecodecamp.org/news/git-best-practices-commits-and-code-reviews/

These short version is this:
1. Make only small changes to each file. Then commit.
2. Name the commits short and sweet.

In this pull request we should have seen a commit at least for every file changed.

We also encountered a merge conflict likely because you didn't maintain a synchronous copy with main. Ensure that you fetch from main so that these conflicts are resolved before they even occur.

When we perform `yarn install` there is some files that will be added to your GitHub as registered changes including the yarn.lock file. Please ensure that these changes are discarded before you commit - they do not need to be present in the repo. 

Going forward in this pull request, @gangamarx can you please conduct application testing only. We are looking to see if each component 1. runs and 2. saves output correctly.

Given you  cant review per commit, please use my below list:

- [ ] src/components/AirbaseNG/AirbaseNG.tsx

- [ ] src/components/CVE-2021-41773/CVE-2021-41773.tsx
- [ ] src/components/CVE-2021-44228/CVE-2021-44228.tsx
- [ ] src/components/CVE-2022-24112/CVE-2022-24112.tsx
- [ ] src/components/Fcrackzip/Fcrackzip.tsx
- [ ] src/components/FindOffset/FindOffset.tsx
- [ ] src/components/Foremost/Foremost.tsx
- [ ] src/components/Gyoithon/Gyoithon.tsx
- [ ] src/components/Hashcat/Hashcat.tsx
- [ ] src/components/Hydra/Hydra.tsx
- [ ] src/components/JohnTheRipper/JohnTheRipper.tsx
- [ ] src/components/Netcat/Netcat.tsx
- [ ] src/components/RTsort/RTsort.tsx
- [ ] src/components/SMBEnumeration/SMBEnumeration.tsx
- [ ] src/components/SearchSploit/SearchSploit.tsx
- [ ] src/components/SnmpCheck/SnmpCheck.tsx
- [ ] src/components/Traceroute/Traceroute.tsx
- [ ] src/components/WPScan/WPScan.tsx
- [ ] src/components/ZeroLogon/Zerologon.tsx
- [ ] src/components/metagoofil/metagoofil.tsx
- [ ] src/components/msfvenom/msfvenom.tsx
- [ ] src/components/theharvester/theharvester.tsx

Once complete we will go through the code itself.

Thank you all for your hard work! Please take some time to go through the resources I gave before your next card is actioned.

Bradley

**Review by bradleybogg — CHANGES_REQUESTED** [review]
See prior comment and please resolve style errors.

**Comment by bradleybogg** [comment]
Failure to update PR closed. Please re open when completed.

**Comment by bradleybogg** [comment]
@CAfitech @hhenryDeakin 

Please review your work against the main changes with the merge conflicts.

Thank you!

**Review by zkkok — COMMENTED** [review]
Hi, @CAfitech @hhenryDeakin ,
Functionality Testing has been conducted. Here is the summary for it
[Functionality Test(SaveOutputv2 Rollout).pdf](https://github.com/Hardhat-Enterprises/Deakin-Detonator-Toolkit/files/15230898/Functionality.Test.SaveOutputv2.Rollout.pdf)

Most of the components have successfully integrated the new changes but some components did meet the expectations (Test#1,#8,#17), please revise your code

**Comment by michaeljpigott** [comment]
Hi @CAfitech please resolve the conflicts and let me know when it is ready for review.

**Comment by CAfitech** [comment]
Ready for review.

**Review by zkkok — APPROVED** [review]
Issues in Test # 1, 8, 17 has been resolved, @michaeljpigott please proceed with the code review

**Comment by michaeljpigott** [comment]
Hi @CAfitech and @hhenryDeakin 

I could not review and approve these changes because it would have created too many conflicts with the baseline bug fixes. I have nearly reviewed and approved most of the baseline bug fixes, so feel free to resolve the conflicts and request my review. 

Kind regards

Michael

**Comment by hhenryDeakin** [comment]
@michaeljpigott 

I have submitted the merge conflict changes and resolved a check fail error by running npx prettier -w metagoofil.tsx. 
These changes are now ready for review and to be merged to main.

Thank you,

Hamilton

File: src/components/CVE-2021-41773/CVE-2021-41773.tsx [hunk 1]
```diff
@@ -3,6 +3,7 @@ import { useForm } from "@mantine/form";
 import { useCallback, useState, useEffect } from "react";
 import { CommandHelper } from "../../utils/CommandHelper";
 import ConsoleWrapper from "../ConsoleWrapper/ConsoleWrapper";
+import { UserGuide } from "../UserGuide/UserGuide";
```
Comment (michaeljpigott) [inline src/components/CVE-2021-41773/CVE-2021-41773.tsx:6]: You do not need UserGuide as it is deprecated and unused. Please delete this line.

**Review by michaeljpigott — APPROVED** [review]
Hi @hhenryDeakin 

I noticed that the UserGuide was added to CVE-2021-41773 when it probably was not required. Otherwise, the pull request looks good. I gave it a brief lookover. 

It would be good if pull requests did not cover so many files in the future because it is hard to review and causes a lot of conflicts. 

Kind regards

Michael

**Pull request merged**